llvm_util/llvm2alive.cpp

```diff
@@ -712,14 +713,35 @@ class llvm2alive_ : public llvm::InstVisitor<llvm2alive_, unique_ptr<Instr>> {
         } else if (name == "align") {
           llvm::Value *ptr = bundle.Inputs[0].get();
           llvm::Value *align = bundle.Inputs[1].get();
-          if (bundle.Inputs.size() != 2)
-            // TODO: "align" assume operand bundle may take a third argument.
-            return error(i);
 
           auto *aptr = get_operand(ptr), *aalign = get_operand(align);
           if (!aptr || !aalign)
             return error(i);
 
+          if (bundle.Inputs.size() >= 3) {
+            assert(bundle.Inputs.size() == 3);
+            auto *adjustofs = get_operand(bundle.Inputs[2].get());
```

Comment:
This is probably guaranteed to be a constant always

Reply:
I think so, I will fix this soon

Reply:
I found that using `get_operand` was simpler... Instead I left an assertion there to make the understanding about align explicit .

Reply:
Can you replace this code with `dyn_cast<Constantint>(..)->getZExtValue()`. It gets rid of the subtraction below. It's a constant; we can negate it here.

Reply:
Its value can be larger than 64 bits (`AlignmentFromAssumptions/simple.ll`).
I'll update this to use `getZExtValue` if its size is not larger than 64 bits.